### ISSUE TITLE
Intermittent `!AreCetShadowStacksEnabled() || UseSpecialUserModeApc()` assertion when embedding CoreCLR

### ISSUE THREAD

**Issue by alexrp** [dotnet/runtime#83437]
### Description

Example failure: https://github.com/vezel-dev/ruptura/actions/runs/4422936308/jobs/7755195387

The TL;DR of this project is that it links a native C DLL with `libnethost.lib` and injects it into a process (in this case, `conhost.exe`). The injected DLL then loads CoreCLR and runs user code.

### Reproduction Steps

It's not going to be easy to create an isolated repro for this one, I think. Here's how to run the failing sample:

```
git clone git@github.com:vezel-dev/ruptura.git
cd ruptura
dotnet example attach
```

(Note that the failure is intermittent.)

### Expected behavior

The sample should run successfully.

### Actual behavior

```
CLR: Assert failure(PID 532 [0x00000214], Thread: 5628 [0x15fc]): !AreCetShadowStacksEnabled() || UseSpecialUserModeApc()
    File: D:\a\_work\1\s\src\coreclr\vm\threads.cpp, Line: 8308 Image:
C:\Windows\SYSTEM32\conhost.exe
```

### Regression?

I *think* it's a regression.

It started happening around the time I upgraded from .NET SDK version 7.0.200 to 7.0.201 and from [runtime.win-x64.Microsoft.NETCore.DotNetAppHost](https://www.nuget.org/packages/runtime.win-x64.Microsoft.NETCore.DotNetAppHost) version 7.0.2 to 7.0.3.

### Configuration

Windows x64

**Comment by janvorli** [comment]
@alexrp I was going to take a look at the repro code and found I am unable to clone your testing repo, I get 
```
git@github.com: Permission denied (publickey).
fatal: Could not read from remote repository.
```

**Comment by alexrp** [comment]
@janvorli hmm, it's a public repo, so might be some SSH key issue on your end? (Just guessing here.)

Maybe try HTTPS instead, i.e. `https://github.com/vezel-dev/ruptura.git`?

**Comment by alexrp** [comment]
I haven't seen this failure in 100+ CI runs over the past 9 months and across various .NET versions (ranging from 7.0.4 to 8.0.0), so I'm inclined to think that whatever caused it has since been fixed. Will go ahead and close this for now.
